In [1]:
# Import our dependencies
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC
from sklearn.metrics import classification_report, confusion_matrix
import pandas as pd
import tensorflow as tf

#  Import and read the diabetes_data.csv.
import pandas as pd
diabetes_df = pd.read_csv('https://raw.githubusercontent.com/ComfyKoala/diabetes-classification/main/Maribel/diabetes_data_upload.csv')
# Display the first few rows
diabetes_df.head()

Age Gender Polyuria Polydipsia sudden weight loss weakness Polyphagia  \
0   40   Male       No        Yes                 No      Yes         No   
1   58   Male       No         No                 No      Yes         No   
2   41   Male      Yes         No                 No      Yes        Yes   
3   45   Male       No         No                Yes      Yes        Yes   
4   60   Male      Yes        Yes                Yes      Yes        Yes   

  Genital thrush visual blurring Itching Irritability delayed healing  \
0             No              No     Yes           No             Yes   
1             No             Yes      No           No              No   
2             No              No     Yes           No             Yes   
3            Yes              No     Yes           No             Yes   
4             No             Yes     Yes          Yes             Yes   

  partial paresis muscle stiffness Alopecia Obesity     class  
0              No              Yes      Yes     Yes  Positive  
1             Yes               No      Yes      No  Positive  
2              No              Yes      Yes      No  Positive  
3              No               No       No      No  Positive  
4             Yes              Yes      Yes     Yes  Positive

In [2]:
#Dropping weakest features "muscle stiffness", "Obesity", "weakness","sudden weight loss","partial paresis"
diabetes_df = diabetes_df.drop(columns=["muscle stiffness", "Obesity", "weakness","sudden weight loss","partial paresis"])
diabetes_df.head(5)

Age Gender Polyuria Polydipsia Polyphagia Genital thrush visual blurring  \
0   40   Male       No        Yes         No             No              No   
1   58   Male       No         No         No             No             Yes   
2   41   Male      Yes         No        Yes             No              No   
3   45   Male       No         No        Yes            Yes              No   
4   60   Male      Yes        Yes        Yes             No             Yes   

  Itching Irritability delayed healing Alopecia     class  
0     Yes           No             Yes      Yes  Positive  
1      No           No              No      Yes  Positive  
2     Yes           No             Yes      Yes  Positive  
3     Yes           No             Yes       No  Positive  
4     Yes          Yes             Yes      Yes  Positive

In [3]:
#change all "yes" to 1 and all "no" to 0
diabetes_df = diabetes_df.replace({"Yes": 1, "No": 0})

#change "female" to 1 and "male" to 0
diabetes_df = diabetes_df.replace({"Female": 1, "Male": 0})

In [4]:
# Get the target variables
target=diabetes_df["class"]
target_names = ["negative", "positive"]

In [5]:
#Get the features
data = diabetes_df.drop("class", axis=1)
feature_names = data.columns
data.head()

Age  Gender  Polyuria  Polydipsia  Polyphagia  Genital thrush  \
0   40       0         0           1           0               0   
1   58       0         0           0           0               0   
2   41       0         1           0           1               0   
3   45       0         0           0           1               1   
4   60       0         1           1           1               0   

   visual blurring  Itching  Irritability  delayed healing  Alopecia  
0                0        1             0                1         1  
1                1        0             0                0         1  
2                0        1             0                1         1  
3                0        1             0                1         0  
4                1        1             1                1         1

In [6]:
# Split data into training and testing
X_train, X_test, y_train, y_test = train_test_split(data, target, random_state=42)

## Support Vector Machines

In [7]:
# Support vector machine linear classifier
model = SVC(kernel='linear')
model.fit(X_train, y_train)

SVC(kernel='linear')

In [8]:
# Model Accuracy
print('Test Acc: %.3f' % model.score(X_test, y_test))

Test Acc: 0.938


In [9]:
# Calculate the classification report
from sklearn.metrics import classification_report
predictions = model.predict(X_test)
print(classification_report(y_test, predictions,
                            target_names=target_names))

              precision    recall  f1-score   support

    negative       0.88      0.96      0.92        46
    positive       0.97      0.93      0.95        84

    accuracy                           0.94       130
   macro avg       0.93      0.94      0.93       130
weighted avg       0.94      0.94      0.94       130



Removed weakest columns and accuracy went from 92.3% in file of "diabetes1updated" to 93.8% above

## Logistic Regression

In [10]:
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression(solver='lbfgs', random_state=1)
classifier

LogisticRegression(random_state=1)

In [11]:
# Train the data
classifier.fit(X_train, y_train)

LogisticRegression(random_state=1)

In [12]:
# Score the model
print(f"Training Data Score: {classifier.score(X_train, y_train)}")
print(f"Testing Data Score: {classifier.score(X_test, y_test)}")

Training Data Score: 0.9358974358974359
Testing Data Score: 0.9307692307692308


In [13]:
# Predict outcomes for test data set
predictions = classifier.predict(X_test)
pd.DataFrame({"Prediction": predictions, "Actual": y_test})

Prediction    Actual
275   Negative  Negative
93    Positive  Positive
6     Positive  Positive
167   Positive  Positive
90    Positive  Positive
..         ...       ...
421   Positive  Positive
182   Positive  Positive
473   Negative  Negative
19    Positive  Positive
397   Negative  Negative

[130 rows x 2 columns]

In [14]:
from sklearn.metrics import accuracy_score
# Display the accuracy score for the test dataset.
accuracy_score(y_test, predictions)

0.9307692307692308